core/src/provider.rs

```diff
@@ -107,43 +111,100 @@ pub struct TraceCallFrame {
+            | NamedChain::Linea => Some(true),
+
+            // Known non-zkEVM chains
+            NamedChain::Mainnet
+            | NamedChain::Sepolia
+            | NamedChain::Arbitrum
+            | NamedChain::Soneium
+            | NamedChain::Avalanche
+            | NamedChain::Polygon
+            | NamedChain::Hyperliquid
+            | NamedChain::Blast
+            | NamedChain::World
+            | NamedChain::Unichain
+            | NamedChain::Base
+            | NamedChain::Optimism
+            | NamedChain::ApeChain
+            | NamedChain::BinanceSmartChain
+            | NamedChain::Fantom
+            | NamedChain::Cronos
+            | NamedChain::Gnosis => Some(false),
+
+            // Fallback for unknown chains
+            _ => None,
+        }
+    } else {
+        None
+    }
+}
+
 impl JsonRpcCachedProvider {
     pub async fn new(
         provider: RindexerProvider,
         chain_id: u64,
         max_block_range: Option<U64>,
     ) -> Self {
-        let response: Result<String, _> = provider.raw_request("zks_L1ChainId".into(), [&()]).await;
-        let is_zk_chain = response.is_ok();
+        let chain = Chain::from(chain_id);
+        let is_zk_evm = match is_known_zk_evm_compatible_chain(chain) {
+            Some(zk) => zk,
+            None => {
+                let response: Result<String, _> =
+                    provider.raw_request("zks_L1ChainId".into(), [&()]).await;
+                let is_zk_chain = response.is_ok();
+                if is_zk_chain {
+                    debug!("Chain {} is zk chain. Trace indexing adjusted if enabled.", chain_id);
+                }
 
-        if is_zk_chain {
-            info!("Chain {} is zk chain. Trace indexing adjusted.", chain_id);
-        }
+                is_zk_chain
+            }
+        };
 
         JsonRpcCachedProvider {
             provider: Arc::new(provider),
             cache: Mutex::new(None),
             max_block_range,
+            chain,
             chain_id,
-            is_zk_chain,
+            is_zk_chain: is_zk_evm,
         }
     }
 
     pub async fn get_latest_block(&self) -> Result<Option<Arc<Block>>, ProviderError> {
         let mut cache_guard = self.cache.lock().await;
+        let block_time =
+            self.chain.average_blocktime_hint().unwrap_or_else(|| Duration::from_millis(1000));
+        let cache_time = block_time;
 
-        // TODO-TODO
+        // Fetches the latest block only if it is likely that a new block has been produced for
+        // this specific network. Consider this to be equal to half the block-time.
         //
-        // This is potentially called way too much. Consider dropping it back.
-        // Also ensure this isn't duplicated by each contract-event.
+        // If we want to reduce RPC calls further at the cost of we could consider indexing delay we
+        // could set this to block-time directly.
         if let Some((timestamp, block)) = &*cache_guard {
```

Comment:
we discussed this on slack to avoid breaking changes or to make rindexer seem slow it should be aggressive by default @jaggad going to make a few changes to allow a blockPollFrequency optional flag in the yaml to config this nicely

Reply:
This has been added now.